src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,814 @@
+    }
+
+    this.#borderSizeRAF = null; // Free the memory
+  };
+
+  #fixDialogPosition = () => {
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+    const inlineStart = this.#isRTLDirection
+      ? document.documentElement.getBoundingClientRect().width -
+        (dialogRect.left + dialogRect.width)
+      : dialogRect.left;
+    const blockStart = dialogRect.top;
+
+    setProperty(
+      this.#dialogRef,
+      DIALOG_INLINE_START,
+      inlineStart - this.#draggedDistanceX
+    );
+    setProperty(
+      this.#dialogRef,
+      DIALOG_BLOCK_START,
+      blockStart - this.#draggedDistanceY
+    );
+  };
+
+  #unfixDialogPosition = () => {
+    this.#dialogRef.style.removeProperty(DIALOG_INLINE_START);
+    this.#dialogRef.style.removeProperty(DIALOG_BLOCK_START);
+  };
+
+  render() {
+    const dialogParts = this.showHeader ? "dialog has-header" : "dialog";
+
+    return (
+      <Host
+        class={{
+          "gx-dialog-header-drag": !this.hidden && this.allowDrag === "header",
+          [RESIZING_CLASS]: this.resizing
+        }}
+      >
+        <dialog
+          aria-labelledby={this.caption ? "heading" : null}
+          part={dialogParts}
+          onClose={this.#handleDialogToggle}
+          ref={el => (this.#dialogRef = el)}
+          onMouseDown={this.allowDrag === "box" ? this.#handleMouseDown : null}
+        >
+          {this.showHeader && (
+            <header
+              class="header"
+              part="header"
+              onMouseDown={
+                this.allowDrag === "header" ? this.#handleMouseDown : null
+              }
+            >
+              <slot name="header-start" />
+              {this.caption && (
+                <h2 id="heading" part="caption" class="caption">
+                  {this.caption}
+                </h2>
+              )}
+              <slot name="header-end" />
+              <button
+                part="close"
+                class="close-button"
+                onClick={this.#closeHandler}
+              ></button>
+            </header>
+          )}
+          <div part="content">
+            <slot />
+          </div>
+          {this.resizable &&
+            !this.hidden && [
+              <div
+                class="edge__block-start"
+                part="dragger edge-block-start"
+                onMouseDown={this.#handleEdgeResize("block-start")}
+              ></div>, // Top
+              <div
+                class="edge__inline-end"
+                part="dragger edge-inline-end"
+                onMouseDown={this.#handleEdgeResize("inline-end")}
+              ></div>, // Right
+              <div
+                class="edge__block-end"
+                part="dragger edge-block-end"
+                onMouseDown={this.#handleEdgeResize("block-end")}
+              ></div>, // Bottom
+              <div
+                class="edge__inline-start"
+                part="dragger edge-inline-start"
```

Comment:
Maybe "edge" instead of "dragger"?

```suggestion
                part="edge block-start"
                onMouseDown={this.#handleEdgeResize("block-start")}
              ></div>, // Top
              <div
                class="edge__inline-end"
                part="edge inline-end"
                onMouseDown={this.#handleEdgeResize("inline-end")}
              ></div>, // Right
              <div
                class="edge__block-end"
                part="edge block-end"
                onMouseDown={this.#handleEdgeResize("block-end")}
              ></div>, // Bottom
              <div
                class="edge__inline-start"
                part="edge inline-start"
```